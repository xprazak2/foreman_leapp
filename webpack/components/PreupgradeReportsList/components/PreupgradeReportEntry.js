import React from 'react';
import { ListView, Grid, Icon } from 'patternfly-react';
import PropTypes from 'prop-types';
import './foreman_leapp.scss';

import InhibitorInfoItem from './InhibitorInfoItem';

import {
  itemIteratorId,
  presentSeverity,
  hasRemediations,
  getExternals,
  getRemediations,
  getTitle,
  getSeverity,
  getSummary,
  getTags,
} from './helpers';

const PreupgradeReportEntry = ({ entry }) => (
  <ListView.Item
    key={entry.id}
    stacked
    checkboxInput={<input type="checkbox" />}
    description={entry.title}
    additionalInfo={[
      <ListView.InfoItem key={itemIteratorId(entry, entry.hostname)}>
        <Icon type="pf" name="cluster" /> {entry.hostname}
      </ListView.InfoItem>,
      <ListView.InfoItem key={itemIteratorId(entry, entry.severity)}>
        <Icon
          type="pf"
          name="resources-full"
          style={{ color: presentSeverity(entry.severity) }}
        />
        {entry.severity}
      </ListView.InfoItem>,
      hasRemediations(entry),
      <InhibitorInfoItem entry={entry} />
    ]}
  >
    <Grid fluid>
      {getTitle(entry)}
      {getSeverity(entry)}
      {getSummary(entry)}
      {getTags(entry)}
      {getExternals(entry)}
      {getRemediations(entry)}
    </Grid>
  </ListView.Item>
);

PreupgradeReportEntry.propTypes = {
  entry: PropTypes.object.isRequired,
};

export default PreupgradeReportEntry;
