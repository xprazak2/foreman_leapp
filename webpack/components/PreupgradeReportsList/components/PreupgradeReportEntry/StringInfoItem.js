import React from 'react';
import EllipsisWithTooltip from 'react-ellipsis-with-tooltip';
import PropTypes from 'prop-types';
import { translate as __ } from 'foremanReact/common/I18n';

import InfoItem from './InfoItem';
import { itemIteratorId } from './helpers';

const StringInfoItem = ({
  entry,
  attr,
  tooltipText,
  translate,
  mapAttr,
  elipsed,
}) => {
  const inner = (
    <span>{translate ? __(mapAttr(entry, attr)) : mapAttr(entry, attr)}</span>
  );
  const innerContent = elipsed ? (
    <EllipsisWithTooltip placement="top">{inner}</EllipsisWithTooltip>
  ) : (
    inner
  );

  return (
    <InfoItem itemId={itemIteratorId(entry, attr)} tooltipText={tooltipText}>
      {innerContent}
    </InfoItem>
  );
};

StringInfoItem.propTypes = {
  entry: PropTypes.object.isRequired,
  attr: PropTypes.string.isRequired,
  tooltipText: PropTypes.string,
  translate: PropTypes.bool,
  mapAttr: PropTypes.func,
  elipsed: PropTypes.bool,
};

StringInfoItem.defaultProps = {
  translate: false,
  mapAttr: (entry, attr) => entry[attr],
  elipsed: false,
  tooltipText: undefined,
};

export default StringInfoItem;
