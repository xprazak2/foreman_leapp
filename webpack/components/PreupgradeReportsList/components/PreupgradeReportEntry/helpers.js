import React from 'react';

import EmptyInfoItem from './EmptyInfoItem';
import StringInfoItem from './StringInfoItem';

export const itemIteratorId = (entry, ...rest) =>
  `${entry.id}-${entry.preupgradeReportId}-${rest.join('-')}`;

export const additionalInfo = entry => {
  const infoAttrs = ['title', 'severity'];

  return infoAttrs.map(attr => {
    const key = itemIteratorId(entry, attr);

    if (!entry[attr]) {
      return <EmptyInfoItem entry={entry} attr={attr} key={key} />;
    }

    switch (attr) {
      case 'title':
        return <StringInfoItem entry={entry} attr={attr} key={key} />;
      case 'severity':
        return <StringInfoItem entry={entry} attr={attr} key={key} />;
      default:
        return '';
    }
  });
};
