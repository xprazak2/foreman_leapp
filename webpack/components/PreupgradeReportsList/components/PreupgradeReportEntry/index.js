import React from 'react';

import { ListView } from 'patternfly-react';
import { additionalInfo } from './helpers';

const PreupgradeReportEntry = props => {
  return (
    <ListView.Item
      key={props.entry.id}
      className="listViewItem--listItemVariants"
      stacked
      additionalInfo={additionalInfo(props.entry)}
    >

    </ListView.Item>
  )
}

export default PreupgradeReportEntry;
