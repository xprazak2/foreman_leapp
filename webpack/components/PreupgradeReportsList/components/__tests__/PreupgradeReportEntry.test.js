import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import PreupgradeReportEntry from '../PreupgradeReportEntry';

const fixtures = {
  'should render': {
    entry: { title: 'no title', severity: 'low', id: 5, preupgradeReportId: 7 },
    isEntrySelected: false,
    toggleSelected: () => {},
  },
};

describe('PreupgradeReportEntry', () =>
  testComponentSnapshotsWithFixtures(PreupgradeReportEntry, fixtures));
