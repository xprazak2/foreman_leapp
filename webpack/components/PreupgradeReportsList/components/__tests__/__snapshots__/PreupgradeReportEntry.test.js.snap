// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreupgradeReportEntry should render 1`] = `
<ListViewItem
  actions={null}
  additionalInfo={
    Array [
      <ListViewInfoItem
        className=""
        stacked={false}
      >
        <Icon
          name="cluster"
          type="pf"
        />
         
      </ListViewInfoItem>,
      <ListViewInfoItem
        className=""
        stacked={false}
      >
        <Icon
          name="resources-full"
          style={
            Object {
              "color": "#39a5dc",
            }
          }
          type="pf"
        />
        low
      </ListViewInfoItem>,
      <EmptyInfoItem
        attr="remediation"
        entry={
          Object {
            "id": 5,
            "preupgradeReportId": 7,
            "severity": "low",
            "title": "no title",
          }
        }
      />,
      <InhibitorInfoItem
        entry={
          Object {
            "id": 5,
            "preupgradeReportId": 7,
            "severity": "low",
            "title": "no title",
          }
        }
      />,
    ]
  }
  checkboxInput={
    <input
      type="checkbox"
    />
  }
  compoundExpand={false}
  compoundExpanded={false}
  description="no title"
  heading={null}
  hideCloseIcon={false}
  initExpanded={false}
  key="5"
  leftContent={null}
  onCloseCompoundExpand={[Function]}
  onExpand={[Function]}
  onExpandClose={[Function]}
  stacked={true}
>
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={true}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        md={1}
      >
        <strong>
          Title
        </strong>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        md={8}
      >
        no title
      </Col>
    </Row>
    <Row
      bsClass="row"
      className="top-padded"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        md={1}
      >
        <strong>
          Risk Factor
        </strong>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        md={8}
      >
        <Icon
          name="resources-full"
          style={
            Object {
              "color": "#39a5dc",
            }
          }
          type="pf"
        />
         
        low
      </Col>
    </Row>
    <div />
    <div />
    <div />
    <div />
  </Grid>
</ListViewItem>
`;
